### PR TITLE
Simplify schemas

### PR DESCRIPTION
**Pull request by RussTorres**
refactor addressing #74 and #71

Adds schemas and argschemaparser classes to rendermodules.module which facilitate common stack operations and spark use cases to allow for code reuse.

Also important -- most tests now create new dictionary objects via copy or dict() prior to running tests as pre_load operations in schemas were changing the dictionary objects being used by other tests.

### CODE REVIEW

File: rendermodules/module/schemas/stack_schemas.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    maxZ = argschema.fields.Int(required=False)
+    z = argschema.fields.Int(required=False)
+    zValues = argschema.fields.List(argschema.fields.Int, required=False)
+
+    @post_load
+    def generate_zValues(self, data):
+        return self._generate_zValues(data)
+
+    def _generate_zValues(self, data):
+        print data
```
Comment (mahaling) [inline rendermodules/module/schemas/stack_schemas.py]: this print statement should be removed

Comment (RussTorres) [reply rendermodules/module/schemas/stack_schemas.py]: Good catch, fixed along with another one I left in.

File: rendermodules/dataimport/apply_mipmaps_to_render.py [hunk 2]
```diff
@@ -74,15 +74,7 @@ def run(self):
         zvalues = self.render.run(
```
Comment (fcollman) [inline rendermodules/dataimport/apply_mipmaps_to_render.py]: I think this is a pretty common functionality that we might implement in the StackTransitionModule or a subclass of it.

File: rendermodules/module/schemas/stack_schemas.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+        if 'zValues' in data:
+            return
+        elif 'z' in data:
+            data['zValues'] = [data['z']]
+        elif ('minZ' in data) and ('maxZ' in data):
+            data['zValues'] = range(data['minZ'], data['maxZ'] + 1)
+        else:
+            raise ValidationError("no z values specified")
+
+    def _override_input(self, data):
```
Comment (fcollman) [inline rendermodules/module/schemas/stack_schemas.py]: can you add a comment with #DEPRECATED to help us find this when we want to get rid of it at some later point.

**Pull request merged**